runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -327,32 +332,37 @@ public LayoutBuilder removeSequencerServers(Set<String> endpoints)
      * Removes the logunit endpoint from the layout.
      *
      * @param endpoint Log unit server to be removed
-     * @return Workflow manager
+     * @return Builder
      * @throws LayoutModificationException Cannot remove non-replicated logunit
      */
     public LayoutBuilder removeLogunitServer(String endpoint)
-            throws LayoutModificationException {
+            throws LayoutModificationException , CloneNotSupportedException{
+
+        Layout tempLayout = (Layout) layout.clone();
 
-        List<LayoutSegment> layoutSegments = layout.getSegments();
+        List<LayoutSegment> layoutSegments = tempLayout.getSegments();
         for (LayoutSegment layoutSegment : layoutSegments) {
             for (LayoutStripe layoutStripe : layoutSegment.getStripes()) {
-
-                List<String> loguintServers = layoutStripe.getLogServers();
-
-                for (int k = 0; k < loguintServers.size(); k++) {
-                    if (loguintServers.get(k).equals(endpoint)) {
-                        if (loguintServers.size() == 1) {
-                            throw new LayoutModificationException(
-                                    "Attempting to remove all logunit. No replicas available.");
+                if(layoutStripe.getLogServers().remove(endpoint)) {
+                    if (layoutStripe.getLogServers().isEmpty()) {
+                        throw new LayoutModificationException(
+                                "Attempting to remove all logunit in stripe. "
+                                        + "No replicas available.");
+                    } else {
+                        if (layoutSegment.getReplicationMode() == Layout.ReplicationMode.CHAIN_REPLICATION) {
+                            if (layoutStripe.getLogServers().size() < 2) {
```

Comment:
should 2 go into some constant or properties file. Although 2 is redundancy so I am not sure !

Reply:
We need a proper way to specify server properties and this seems like one of those properties.
I'll make it a constant for now.